app/models/service_template_transformation_plan_task.rb

```diff
@@ -34,9 +34,74 @@ def task_active
     vm_resource.update_attributes(:status => ServiceResource::STATUS_ACTIVE)
   end
 
+  def conversion_host
+    Host.find_by(:id => options[:transformation_host_id])
+  end
+
+  def transformation_log
+    host = conversion_host
+    if host.nil?
+      msg = "Conversion host was not found: ID [#{options[:transformation_host_id]}]. Download of transformation log aborted."
+      _log.error(msg)
+      raise MiqException::Error, msg
+    end
+
+    userid, password = host.auth_user_pwd(:remote)
+    if userid.blank? || password.blank?
+      msg = "Host credential is required: userid [#{userid}], password [#{password}]. Download of transformation log aborted."
+      _log.error(msg)
+      raise MiqException::Error, msg
+    end
+
+    logfile = options.fetch_path(:virtv2v_wrapper, "v2v_log")
+    if logfile.blank?
+      msg = "The location of transformation log was not set. Download of transformation log aborted."
+      _log.error(msg)
+      raise MiqException::Error, msg
+    end
+
+    begin
+      require 'net/scp'
+      Net::SCP::download!(host.ipaddress, userid, logfile, nil, :ssh => {:password => password})
+    rescue => scp_err
+      _log.error("Download of transformation log for #{description} with ID [#{id}] failed with error: #{scp_err.message}")
+      raise scp_err
+    end
+  end
+
+  # Intend to be called by UI to display transformation log. The log is stored in MiqTask#task_results
+  # Since the task_results may contain a large block of data, it is desired to remove the task upon receiving the data
+  def transformation_log_queue(userid)
```

Comment:
Seems like we should figure out `userid` up front and then pass that into `create_error_status_task` or `queue_options` instead of making both do the `|| 'system'` logic.

What do you think about `userid` being set set from `User.current_user` instead of being passed in?  If we keep the parameter I suggest defaulting it since you are handling the 'nil' case.